types/time.go

```diff
@@ -2432,12 +2432,29 @@ func microSeconds(t *MysqlTime, input string, ctx map[string]int) (string, bool)
 }
 
 func yearNumericFourDigits(t *MysqlTime, input string, ctx map[string]int) (string, bool) {
-	v, succ := parseDigits(input, 4)
+	return yearNumericNDigits(t, input, ctx, 4)
+}
+
+func yearNumericTwoDigits(t *MysqlTime, input string, ctx map[string]int) (string, bool) {
+	return yearNumericNDigits(t, input, ctx, 2)
+}
+
+func yearNumericNDigits(t *MysqlTime, input string, ctx map[string]int, n int) (string, bool) {
+	var l int
+	for l = 1; l < n+1; l++ {
+		if _, succ := parseDigits(input, l); !succ {
+			break
+		}
+	}
+	v, succ := parseDigits(input, l-1)
```

Comment:
Reaching here, I think `parseDigits(input, l-1)` is guaranteed to be `true`?

Reply:
I think if the first byte of the year str is not a digit, like
```
select str_to_date('a69/10/22','%Y/%m/%d')
```
 `parseDigits(input, l-1)`  will not success...It become  `parseDigits(input, 0)`